src/app/core/services/auth.service.ts

```diff
@@ -102,6 +102,15 @@ export class AuthService {
     return this.hatSvc.getApplicationHmi()
       .pipe(map((apps: HatApplication[]) => {
         const parentApp = apps.find(app => app.application.id === parentAppId);
+
+        if (!parentApp) {
+          throw new Error('application_id_not_found ');
+        }
+
+        if (parentApp.application.kind.kind !== 'App') {
```

Comment:
can be merged into single `if (!parentApp || parentApp.application.kind.kind !== 'App')`